packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts

```diff
@@ -194,18 +199,34 @@ export class NetworkLoadBalancer extends BaseLoadBalancer implements INetworkLoa
   }
 
   public readonly metrics: INetworkLoadBalancerMetrics;
-  public readonly securityGroups?: string[];
   public readonly ipAddressType?: IpAddressType;
+  public readonly connections: ec2.Connections;
+  /**
+   * The security groups that were passed in Props.
+   */
+  private readonly originalSecurityGroups?: ec2.ISecurityGroup[];
+  public get securityGroups() {
+    /**
+     * With the implementation of IConnectable, connections security groups are now used instead of the security group passed in props,
+     * however, since connections always return arrays that are not undefined and NLB does not allow mutual changes of undefined and empty array,
+     * when the connections security groups is an empty array, the original security group is specified for backwards compatible. (original can be undefined or empty array)
+     * https://github.com/aws/aws-cdk/pull/28494
+     */
+    return this.connections.securityGroups.length > 0
+      ? this.connections.securityGroups.map(sg => sg.securityGroupId)
+      : this.originalSecurityGroups?.map(sg => sg.securityGroupId);
```

Comment:
```suggestion
    return this.connections.securityGroups.length > 0
      ? this.connections.securityGroups.map(sg => sg.securityGroupId)
      : this.isSecurityGroupsPropertyDefined ? [] : undefined;
```
We'll reach the second condition only to return `[]` or `undefined` (correct me if I'm wrong).
I wonder if using a boolean makes for a clearer implementation.
What do you think?

Reply:
Exactly! this is nice!